idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,140 @@
+        """ Reads in raw amr results file outputted by srst2, and does initial processing of marking gene family,
+        + adding in blank columns for relevant metrics (total_genes_hit, total_coverage, total_depth)."""
+        amr_results = pd.read_csv(amr_raw_path, delimiter="\t")
+        # Parse out gene family as substring after '_', e.g. Aph_AGly's gene family would be AGly
+        amr_results['gene_family'] = amr_results.apply(lambda row: row.gene[row.gene.index('_')+1:], axis=1)
+        # Add empty columns for later processing
+        amr_results['total_genes_hit'], amr_results['total_coverage'], amr_results['total_depth'] = (
+            [amr_results.apply(lambda _: '', axis=1)]) * 3
+        return amr_results
+
+    @staticmethod
+    def _summarize_amr_gene_families(amr_raw_path):
+        """Returns a gene family-level summary of total_genes_hit, total_coverage, and total_depth."""
+        amr_results = PipelineStepRunSRST2._get_pre_proc_amr_results(amr_raw_path)
+        amr_results_by_gene_fam = amr_results.groupby(['gene_family'])
+        total_genes_hit, total_coverage, total_depth = (amr_results_by_gene_fam.size().reset_index(name='total_genes_hit'),
+                                                        amr_results_by_gene_fam['coverage'].sum().reset_index(name='total_coverage'),
+                                                        amr_results_by_gene_fam['depth'].sum().reset_index(name='total_depth')
+                                                       )
+        amr_summary = pd.merge(pd.merge(total_genes_hit, total_coverage, on='gene_family', how='left'),
+                               total_depth, on='gene_family', how='left')
+        return amr_summary
+
+    @staticmethod
+    def _get_loc(df, row_i, col_name):
+        """Helper function to get a pandas df location given a row index and col name."""
+        return df.iloc[row_i, df.columns.get_loc(col_name)] 
+
+    @staticmethod
+    def _assign_locations(df1, row_idx1, df2, row_idx2, *col_names):
+        """ Assigns values of col_names in df2 to the same col_names in df1.
+            Assumes df1 and df2 contain the same col_names. """
+        for col_name in col_names:
+            df1.iloc[row_idx1, df1.columns.get_loc(col_name)] = PipelineStepRunSRST2._get_loc(df2, row_idx2, col_name)
+    
+    @staticmethod
+    def _process_amr_results(amr_results_path):
+        """ Returns processed amr result table with total_genes_hit, total_coverage,
+            and total_depth column values filled in for all genes. """
+        amr_summary = PipelineStepRunSRST2._summarize_amr_gene_families(amr_results_path)
+        amr_results = PipelineStepRunSRST2._get_pre_proc_amr_results(amr_results_path)
+        # Sorting reduces run time when amalgamating amr_results and amr_summary
+        sorted_amr = amr_results.sort_values(by=['gene_family'])
+        i_srt_amr_row = 0
+        for i_amr_smry_row in range(amr_summary.shape[0]):
```

Comment:
In general, you shouldn't iterate over rows of a dataframe. Pandas probably has a more efficient way of doing it. In this case, just do a merge (left outer join) of the long dataframe (amr_results) with the short dataframe (amr_summary).